### ISSUE TITLE
Assume first event message field required

### ISSUE THREAD

**Issue by alexander-yevsyukov** [SpineEventEngine/core-java#896]
We have a convention of requiring first command message field. Similarly, an event message should assume that the first field is an ID of the entity which generated the event, and as such, is required.

**Comment by alexander-yevsyukov** [comment]
@armiol, or we can have a better convention which makes all fields which name ends with `_id` required. And they can be made optional with `(required) = false`.

**Comment by armiol** [comment]
Taking #905 into considerations.

@alexander-yevsyukov Maybe we could make `_id` fields required _and_ use them for routing?

**Comment by alexander-yevsyukov** [comment]
@armiol, I think doing both is twice better :)

**Comment by armiol** [comment]
As a result of the discussion, it was decided that the `event` fields should not be required. We now have two common scenarios:

1. Routing from `Aggregate` and` ProcessManager` to `Projection`s by` producer_id`. Typical case is displaying the portion of a write-side entity on UI. In this case, `producer_id` is already populated, and the first field of the event may easily be optional.

2. Routing according to the `event` field value. It's either automatic (the framework itself chooses the event field by its type to match the entity ID type) or custom (the developer sets the logic). In this case, any event field can be either `required` or optional.

Routing of external events are reduced to p.2.

In addition, the proposal to name the event fields with the suffix `_id` goes against the principle we advocate, i.e. write and read less code:
```proto
message Movie {

    // ...

    PublisherId publisher = 7; // not `publisher_id`.
}
```
In short, if we want, we’ll introduce any bells and whistles later.
